src/librustc/hir/check_attr.rs

```diff
@@ -154,7 +154,22 @@ impl<'a, 'tcx> CheckAttrVisitor<'a, 'tcx> {
         let hints: Vec<_> = item.attrs
             .iter()
             .filter(|attr| attr.name() == "repr")
-            .filter_map(|attr| attr.meta_item_list())
+            .filter_map(|attr| {
+                let list = attr.meta_item_list();
+                let mut has_hints = false;
+                if let Some(ref list) = list {
+                    has_hints = !list.is_empty();
+                }
+                if !has_hints {
+                    span_warn!(
```

Comment:
This will be an unsilencable warning instead of a lint, is that intended? So far we reserved that for future incompatibility warnings.

A test for this would be good, too

Reply:
I'll be adding a test. I feel that an empty `repr` (or it's cousin, `#[repr = "C"]`) should be hard failures in the future. What do you think?

Reply:
I don't think all the overhead of carefully managing the transition to a hard error, and the eventual breakage of (hypothetical) old crates, is worth it for this. This is just supposed to catch a silly programmer mistake, it's not a soundness issue or anything. IMO this should be a lint, warn-by-default for now and maybe later upgraded to deny-by-default.

Reply:
I feel strongly that having an incorrect `repl` (specially `#[repr = "C"]`) is hiding a pretty severe bug. Having said that, I can see how it should be a lint instead. I'll push my latest changes and rework this into a lint at a later time.

Reply:
Will turn into a lint once I free myself to work on this a bit more. Please verify that the current output is reasonable.